File: backend/RadianciaKS.Domain/Models/StoreSettings.cs
namespace RadianciaKS.Domain.Models
{
    public class StoreSettings : EntityBase
    {
        public string StoreName { get; set; } = string.Empty;
        public string CNPJ { get; set; } = string.Empty;
        public string? Address { get; set; }
        public string? Phone { get; set; }
        public string? ReceiptFooter { get; set; }
        public string? SmallLogoPath { get; set; }
        public string? BigLogoPath { get; set; }
        public decimal ServiceCharge { get; set; }

        public void Update(string storeName,
        string cnpj, string? address,
        string? phone, string? receiptFooter,
        string? smallLogoPath, string? bigLogoPath,
        decimal serviceCharge)
        {
            StoreName = storeName;
            CNPJ = cnpj;
            Address = address;
            Phone = phone;
            ReceiptFooter = receiptFooter;
            SmallLogoPath = smallLogoPath;
            BigLogoPath = bigLogoPath;
            ServiceCharge = serviceCharge;
        }
    }
}

File: frontend/src/app/features/backoffice/pages/store-settings/store-settings.ts
import { CommonModule } from '@angular/common';
import { Component, inject, OnInit } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { ButtonComponent } from '../../../../shared/components/button-component/button-component';
import { InputComponent } from '../../../../shared/components/input-component/input-component';
import { StoreSettingsRequestDto } from '../../../../core/models/store-settings.model';
import { StoreSettingsService } from '../../../../core/services/store-settings-service';

@Component({
  selector: 'app-store-settings',
  imports: [CommonModule, FormsModule, ButtonComponent, InputComponent],
  templateUrl: './store-settings.html',
  styleUrl: './store-settings.scss',
})
export class StoreSettings implements OnInit {
  private storeService = inject(StoreSettingsService);
  settings: StoreSettingsRequestDto = {
    storeName: '',
    cnpj: '',
    address: '',
    phone: '',
    receiptFooter: '',
    smallLogoPath: '',
    bigLogoPath: '',
    serviceCharge: 0,
  };

  ngOnInit() {
    this.storeService.getSettings().subscribe((data) => {
      this.settings = { ...data };
    });
  }

  onLogoSelected(event: any, isBig: boolean): void {
    const file = event.target.files[0];
    if (file) {
      this.storeService.uploadLogo(file, isBig).subscribe((res) => {
        if (isBig) this.settings.bigLogoPath = res.url;
        else this.settings.smallLogoPath = res.url;
      });
    }
  }

  save() {
    this.storeService.updateSettings(this.settings).subscribe(() => alert('Configurações salvas!'));
  }
}
